control-plane-operator/controllers/hostedcontrolplane/ocm/config.go

```diff
@@ -51,9 +52,58 @@ func reconcileConfig(cfg *openshiftcpv1.OpenShiftControllerManagerConfig, deploy
 		Kind:       "OpenShiftControllerManagerConfig",
 		APIVersion: openshiftcpv1.GroupVersion.String(),
 	}
+
 	cfg.Build.ImageTemplateFormat.Format = dockerBuilderImage
 	cfg.Deployer.ImageTemplateFormat.Format = deployerImage
-	cfg.DockerPullSecret.InternalRegistryHostname = config.DefaultImageRegistryHostname
+
+	// registry config
+	cfg.DockerPullSecret.InternalRegistryHostname = imageConfig.Status.InternalRegistryHostname
+	cfg.DockerPullSecret.RegistryURLs = imageConfig.Status.ExternalRegistryHostnames
+	if len(cfg.DockerPullSecret.InternalRegistryHostname) == 0 {
+		cfg.DockerPullSecret.InternalRegistryHostname = config.DefaultImageRegistryHostname
+	}
+
+	// build config
+	if hasBuildDefaults(buildConfig) {
```

Comment:
Any way we can import this configv1.BuildConfig -> OCM.Build translation rather than duplicate it? This seems like something that is easy to fall out of sync in the future

Reply:
The way this is done in the openshift controller manager operator today is via mapping of unstructured fields:
https://github.com/openshift/cluster-openshift-controller-manager-operator/blob/efd1aeb288e8c45a3cfb75483e1b24830492063b/pkg/operator/configobservation/builds/observe_builds.go#L76-L112
Besides not being easy to vendor, it doesn't provide early feedback that something is broken when your go types change.